src/main/java/io/jenkins/plugins/intotorecorder/transport/Grafeas.java

```diff
@@ -0,0 +1,177 @@
+            }
+        }
+
+        public GrafeasInTotoMetadata(Link link) {
+            this.signatures = link.getSignatures();
+            this.signed = new GrafeasInTotoLink(
+                link.getCommand(),
+                link.getMaterials(),
+                link.getProducts(),
+                link.getByproducts(),
+                link.getEnvironment()
+            );
+        }
+    }
+
+
+    public class GrafeasOccurrence {
+        public String noteName;
+        public Map<String, String> resource = new HashMap<String, String>();
+        public String kind = "INTOTO";
+        public GrafeasInTotoMetadata intoto;
+
+        public GrafeasOccurrence(String noteName, String resourceUri) {
+            this.noteName = noteName;
+            this.resource.put("uri", resourceUri);
+        }
+    }
+
+    private static Map<String, String> getParameterMap(String parameterString) {
+        String[] items = parameterString.split("&");
+        Map<String, String> parameterMap = new HashMap<String, String>();
+        for (String item : items) {
+            String[] pair = item.split("=");
+            parameterMap.put(pair[0], pair[1]);
+        }
+        return parameterMap;
+    }
+
+    public Grafeas(URI uri) {
+        String scheme = uri.getScheme().split("\\+")[1];
+        String authority = uri.getAuthority();
+        String path = uri.getPath();
+        URIBuilder uriBuilder = new URIBuilder();
+
+        try {
+            this.uri = uriBuilder
+                .setScheme(scheme)
+                .setHost(authority)
+                .setPath(path)
+                .build();
+        } catch (URISyntaxException e) {
+            throw new RuntimeException("unable to build Grafeas URI: " + e.toString());
+        }
+
+        String parameterString = uri.getQuery();
+
+        Map<String, String> parameterMap = this.getParameterMap(parameterString);
+
+        GrafeasOccurrence occurrence = new GrafeasOccurrence(
+            parameterMap.get("noteName"),
+            parameterMap.get("resourceUri")
+        );
+
+        this.occurrence = occurrence;
+    }
+
+    public void submit(Link link) {
+        this.occurrence.intoto = new GrafeasInTotoMetadata(link);
+
+        Gson gson = new Gson();
+        String jsonString = gson.toJson(this.occurrence);
+        GenericUrl url = new GenericUrl(this.uri);
+
+        // FIXME: Shamelessly copied from GenericCRUD.java
+        try {
+            HttpRequest request = new NetHttpTransport()
+                .createRequestFactory()
+                .buildPostRequest(url,
+                    ByteArrayContent.fromString("application/json",
+                        jsonString));
+            HttpResponse response = request.execute();
+            System.out.println(response.parseAsString());
+
+            /* FIXME: should handle error codes and other situations more appropriately,
+             * but this gets the job done for a PoC
+             */
```

Comment:
can we ticketize this FIXME?

Reply:
https://github.com/jenkinsci/in-toto-plugin/issues/14